openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -387,6 +392,35 @@ def finish_auth(request):
     })
 
 
+def enterprise_selection_page(request, user, next_url):
+    """
+    Updates redirect url to enterprise selection page if user is associated
+    with multiple enterprises otherwise return the next url.
+
+    param:
+      next_url(string): The URL to redirect to after multiple enterprise selection or in case
+      the selection page is bypassed e.g when dealing with direct enrolment urls.
+    """
+    redirect_url = next_url
+
+    response = get_enterprise_learner_data_from_api(user)
+    if response and len(response) > 1:
+        redirect_url = reverse('enterprise_select_active') + '/?success_url=' + next_url
+
+        # Check to see if next url has an enterprise in it. In this case if user is associated with
+        # that enterprise, activate that enterprise and bypass the selection page.
+        if re.match(ENTERPRISE_ENROLLMENT_URL_REGEX, six.moves.urllib.parse.unquote(next_url)):
+            enterprise_in_url = re.search(UUID4_REGEX, next_url).group(0)
+            for enterprise in response:
+                if enterprise_in_url == str(enterprise['enterprise_customer']['uuid']):
+                    is_activated_successfully = activate_learner_enterprise(request, user, enterprise_in_url)
+                    if is_activated_successfully:
+                        redirect_url = next_url
+                    break
```

Comment:
I think this break statement is not needed.

Reply:
we want to make sure that once the enterprise matches the `enterprise_in_url`, we don't keep on looping over the enterprises in the response